source/draw.cpp

```diff
@@ -339,4 +353,776 @@ void draw_vehicle_attitude(i16 alpha, i16 beta, char blink)
+            }
+            rectrel (-2, 0, 4, 0.5, 0.5, 1);
+            rectrel (2, 0, 4, 0.5, 0.5, 1);
+            break;
+        // Orbital engine
+        case 1:
+            for (   a = static_cast<int>((SDL_GetTicks()/tickrate)%45);
+                    a <= 135+(int)((SDL_GetTicks()/tickrate)%45);
+                    a += 45) {
+                _ox = 25 * tcos[a]; _oy = 25 * tsin[a];
+                rel (-_ox, 0, _oy, _ox, 0, -_oy);
+            }
+            break;
+        // Record player
+        case 2:
+            b = 0;
+            if (/* sbf_stat&& */pix==pixel_sonante)
+                b = ((SDL_GetTicks()/tickrate)%3)*15;
+            for (a=b; a<=135+b; a+=45) {
+                _ox = 5 * tcos[a]; _oy = 5 * tsin[a];
+                rel (-_ox, 0, _oy, _ox, 0, -_oy);
+            }
+            rectrel (0, 0, 0, 6, 6, 1);
+            if (recfile) Txt ("> RECORD >", -5.5, 0, -5, 0.1, 0.15, 270, 0);
+            if (globalvocfile[0]!='.') Txt ("> PLAY >", -5.5, 0, -4, 0.1, 0.15, 270, 0);
+            sprintf (t, "INPUT: %s", source_name[static_cast<int>(source)]);
+            Txt (t, -5.5, 0, 5, 0.1, 0.15, 270, 0);
+            sprintf (t, "ACCURATEZZA: %s", record_qlty[static_cast<int>(quality)]);
+            Txt (t, -5.5, 0, 4, 0.1, 0.15, 270, 0);
+            if (repeat) Txt ("REPEAT", -5.5, 0, 3, 0.1, 0.15, 270, 0);
+            break;
+        
+        // other objects
+        // TODO: is this code reading from a file on each tick? Aren't objects loaded into memory on startup?
+        default:
+            Pixel (tipo+FRONTIER_M3);
+            if (id) break;
+            if (!type_mode) break; // ctrlkeys[0]&16 is scroll lock, use something else
+            if (!memcmp(&subsignal[9*(tipo+FRONTIER_M3)], "TEXT-" /*"TESTO-"*/, 5/*6*/)) {
+                a = subsignal[9*(tipo+FRONTIER_M3)+5/*6*/];
+                memset (buffer, ' ', 512);
+                // safeguard buffer from reading beyond the end of the text
+                buffer[512] = '\0';
+                sprintf (autore_forme, "TEXT-%03d.FIX", tipo-3);
+                th = std::fopen (autore_forme, "r+b");
+                if (th) {
+                    buffer[32] = '\0';
+                    k1 = 5.32;
+                    int bytes_read = std::fread(&buffer[33], 1, 512, th);
+                    for (c=0; c<bytes_read; c+=32) {
+                        memcpy (buffer, buffer+33+c, 32);
+                        if (a=='V') {
+                            Txt (reinterpret_cast<char*>(buffer), -6.82, -k1, 0, 0.11, 0.14, 0, 0);
+                        } else if (a=='H') {
+                            Txt (reinterpret_cast<char*>(buffer), -6.82, 0, k1, 0.11, 0.14, 270, 0);
+                        }
+                        k1 -= 0.7;
+                    }
+                    if (d<15||tipo==carry_type) {
+                        fermo_li = 1;
+                        memcpy (buffer+545, buffer+33, 512);
+                        /*
```

Comment:
I'll note that some of these functions may produce sounds, so it may become a bit weird to have `draw.cpp` depending on the audio module. Unless we can reorganize this in some other way, we might have to make to with this.